pluginlib/cmake/pluginlib_enable_plugin_testing.cmake

```diff
@@ -0,0 +1,223 @@
+
+  # Install package.xml, renaming to just `package.xml` if needed
+  set(fake_install_dir "${prefix}/share/${ARG_PACKAGE_NAME}/")
+  list(APPEND input_files "${ARG_PACKAGE_XML}")
+  list(APPEND input_content "")
+  list(APPEND output_files "${fake_install_dir}/package.xml")
+
+  # Add blank file indicating existance of package in the ament index
+  list(APPEND input_files "")
+  list(APPEND input_content "")
+  list(APPEND output_files "${prefix}/share/ament_index/resource_index/packages/${ARG_PACKAGE_NAME}")
+
+  # Write the plugin descriptions in share/
+  set(ament_index_plugin_content "")
+  foreach(plugin_description_path "${ARG_PLUGIN_DESCRIPTIONS}")
+    list(APPEND input_files "${plugin_description_path}")
+    list(APPEND input_content "")
+    list(APPEND output_files "${prefix}/share/${ARG_PACKAGE_NAME}/${plugin_description_path}")
+
+    set(ament_index_plugin_content "${ament_index_plugin_content}\nshare/${ARG_PACKAGE_NAME}/${plugin_description_path}")
+  endforeach()
+  # Get rid of leading newline
+  string(STRIP "${ament_index_plugin_content}" ament_index_plugin_content)
+
+  # Write one file that says where all of this package's plugin description files are
+  list(APPEND input_files "")
+  list(APPEND input_content "${ament_index_plugin_content}")
+  list(APPEND output_files "${prefix}/share/ament_index/resource_index/${ARG_PLUGIN_CATEGORY}__pluginlib__plugin/${ARG_PACKAGE_NAME}")
+
+  # Get rid of FOOBAR
+  list(REMOVE_AT input_files 0)
+  list(REMOVE_AT input_content 0)
+  list(REMOVE_AT output_files 0)
+
+  #####
+  # Create commands to generate mock install space at build time
+  #####
+
+  list(LENGTH output_files num_output_files)
+  # subtract 1 to avoid looping out of range
+  math(EXPR range_stop "${num_output_files} - 1")
+  foreach(idx RANGE ${range_stop})
+    list(GET input_files ${idx} input_file)
+    list(GET input_content ${idx} content)
+    list(GET output_files ${idx} output_file)
+
+    # Make the directory at configure time because configure_file() does not make directories
+    get_filename_component(output_dir "${output_file}" DIRECTORY)
+    file(MAKE_DIRECTORY "${output_dir}")
+
+    if(IS_ABSOLUTE "${input_file}")
+      set(input_file_abs "${input_file}")
+    else()
+      set(input_file_abs "${CMAKE_CURRENT_SOURCE_DIR}/${input_file}")
+    endif()
+
+    # Copy or write file contents at build time
+    if("${input_file}" STREQUAL "")
+      set(FILE_CONTENT "${content}")
+      configure_file("${PLUGINLIB_ENABLE_PLUGIN_TESTING_DIR}/blank.in" "${output_file}" @ONLY)
+    else()
+      configure_file("${input_file_abs}" "${output_file}" COPYONLY)
+    endif()
+  endforeach()
+
+  # Copy library targets at build time; but make directory at configure time
+  # because `cmake -E copy` does not make directories
+  file(MAKE_DIRECTORY "${prefix}/lib")
+  foreach(plugin_library "${ARG_PLUGIN_LIBRARIES}")
+    set(output_library "${prefix}/lib/$<TARGET_FILE:${plugin_library}>")
```

Comment:
@sloretz unused?

Reply:
Oops, yup. Opened PR to remove it on `ros2` branch; will include it into the backport when that's merged.

https://github.com/ros/pluginlib/pull/211